nixos/modules/hardware/intel-sgx.nix

```diff
@@ -0,0 +1,25 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.hardware.cpu.intel.sgx;
+in
+{
+  options = {
+    hardware.cpu.intel.sgx = {
```

Comment:
I think it would make sense to move this option to the `intel-microcode.nix` module and rename it to `intel.nix`.

Reply:
Done